### PR TITLE
Add Hamming exercise.

### PR DESCRIPTION
**Pull request by stevejb71**

### CODE REVIEW

**Comment by paf31** [comment]
Looks great, thanks!

I think there is a one liner version too:

```purescript
hamming xs ys = guard (length xs == length ys) $> length (filter id (zipWith (/=) xs ys))
```

😄

File: exercises/hamming/examples/src/Hamming.purs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import Prelude
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Tuple (Tuple(..))
+import Data.String (uncons)
+
+distance :: String -> String -> Maybe Int
+distance dna1 dna2 = case Tuple (uncons dna1) (uncons dna2) of
+  Tuple (Just {head: h1, tail: t1}) (Just {head: h2, tail: t2}) -> (if h1 /= h2 then ((+) 1) else id) <$> distance t1 t2
+  Tuple Nothing Nothing -> Just 0
+  Tuple _ _ -> Nothing
```
Comment (lpil) [inline exercises/hamming/examples/src/Hamming.purs]: End of file newline is missing here.

Could this have been written with multiple function clauses?

File: exercises/hamming/test/Main.purs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+-- Test/exercise version: "1.0.0"
```
Comment (lpil) [inline exercises/hamming/test/Main.purs]: What's this for?

**Comment by stevejb71** [comment]
I put a comment with the version number from the canonical data in. It should help keep track if new common tests are added, or the spec for the exercise is changed. 

A solution using ```zipWith``` similar to @paf31's would be neater. Do you want me to update it?

**Comment by lpil** [comment]
I think the solution is good :)

It's the version comment an​ established convention in exercism?

**Comment by stevejb71** [comment]
Some streams look to be doing this:
https://github.com/exercism/discussions/issues/133

I maintain the ocaml stream and have added versions to a majority of tests.

**Comment by lpil** [comment]
Seems like a good addition then. Thank you :)

**Pull request merged**